packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/arrow.dart

```diff
@@ -0,0 +1,280 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/material.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+enum ArrowType { up, left, right, down }
+
+/// Widget that wraps and center another widget horizontally with bidirectional arrow (left & right)
+@immutable
+class BidirectionalHorizontalArrowWrapper extends StatelessWidget {
+  const BidirectionalHorizontalArrowWrapper({
+    Key key,
+    this.arrowColor = Colors.white,
+    this.arrowHeadSize = 16.0,
+    this.arrowStrokeWidth = 2.0,
+    @required this.child,
+  })  : assert(child != null),
+        super(key: key);
+
+  final Color arrowColor;
+  final double arrowHeadSize;
+  final double arrowStrokeWidth;
+  final Widget child;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.center,
+      crossAxisAlignment: CrossAxisAlignment.center,
+      children: <Widget>[
+        Expanded(
+          child: ArrowWidget(
+            color: arrowColor,
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.left,
+          ),
+        ),
+        Container(
+          child: child,
+          margin: const EdgeInsets.symmetric(horizontal: 8.0),
+        ),
+        Expanded(
+          child: ArrowWidget(
+            color: arrowColor,
+            headSize: arrowHeadSize,
+            strokeWidth: arrowStrokeWidth,
+            type: ArrowType.right,
+          ),
+        ),
+      ],
+    );
+  }
+}
+
+/// Widget that wraps and center another widget vertically with bidirectional arrow (up & down)
+@immutable
+class BidirectionalVerticalArrowWrapper extends StatelessWidget {
```

Comment:
nit: add a base class
`BidirectionalArrow` 
that does all the work with the subclasses just indicating whether the arrow is vertical or horizontal.

Reply:
it looks like this logic is basically the same as the horizontal case just with a Column instead of a row.
You can use a Flex instead of a Row/Column in the build method to make it easier to reuse.